pkg/transformers/resolver.go

```diff
@@ -0,0 +1,595 @@
+		// and MR list kind objects as the reference targets.
+		// Such expressions are the primary sources of cross API-group
+		// import statements.
+		// Cross API-group extractors are rare, and they should be
+		// handled when they're being added, this transformer does not
+		// consider them.
+		case *ast.KeyValueExpr:
+			// check if the key is "To" and the value is a CompositeLit
+			if key, ok := x.Key.(*ast.Ident); ok && key.Name == "To" {
+				// prevent a previous GVK from being reused
+				group, version, kind, listKind = "", "", "", ""
+				if cl, ok := x.Value.(*ast.CompositeLit); ok {
+					// check if there are any package qualifiers in the CompositeLit
+					for _, elt := range cl.Elts {
+						if kv, ok := elt.(*ast.KeyValueExpr); ok {
+							if uexpr, ok := kv.Value.(*ast.UnaryExpr); ok {
+								if cl, ok := uexpr.X.(*ast.CompositeLit); ok {
+									// then the reference target resides in another API group
+									// and the composite literal is a selector expression such as
+									// v1beta1.MR. In this case, we deduce the GV of the MR and
+									// list using the selector expression and the corresponding
+									// import statements (with the name as the expression).
+									// Kind and list kind are determined from the field selector.
+									key := kv.Key.(*ast.Ident).Name
+									if sexpr, ok := cl.Type.(*ast.SelectorExpr); ok {
+										if ident, ok := sexpr.X.(*ast.Ident); ok {
+											path := importsUsed[ident.Name].path
+											importsUsed[ident.Name] = importUsage{
+												path: path,
+												used: false,
+											}
+											// we will parse the import path such as:
+											// github.com/upbound/provider-aws/apis/ec2/v1beta1
+											// and extract the GV information from it.
+											tokens := strings.Split(path, "/")
+											// e.g., v1beta1
+											version = tokens[len(tokens)-1]
+											// e.g., ec2.aws.upbound.io
+											group = fmt.Sprintf("%s.%s", tokens[len(tokens)-2], apiGroupSuffix)
+											// extract the kind and list kind names from the field
+											// selector.
+											if sexpr.Sel != nil {
+												if key == "List" {
+													listKind = sexpr.Sel.Name
+												} else {
+													kind = sexpr.Sel.Name
+												}
+											}
+										}
+									} else {
+										// then the reference target is in the same package as the
+										// source. We still transform it for uniformity and
+										// in the future, the source and target might still be
+										// moved to different packages.
+										// The GV information comes from file name in this case:
+										// apis/cur/v1beta1/zz_generated.resolvers.go
+										tokens := strings.Split(filePath, "/")
+										// e.g., v1beta1
+										version = tokens[len(tokens)-2]
+										// e.g., cur.aws.upbound.io
+										group = fmt.Sprintf("%s.%s", tokens[len(tokens)-3], apiGroupSuffix)
```

Comment:
Is there the above problem here?

Reply:
Yes, we need to address this also. Please see the above comment explaining group overrides configuration and the test with `upbound/provider-azure`. Thank you @sergenyalcin.